### DISCUSSION TITLE
Xamarin forms, Azure backend and Dotmim

### DISCUSSION THREAD

**Discussion by kmostofi** [Mimetis/Dotmim.Sync#583] [Q&A]
Hi,

I am new to this project and would like to use it in my project. However, I am confused about how to set it up for a Xamarin form project that uses Azure backend and local Sqlite. 

Do I need to run the script to set up the necessary columns (tracking)? I have the following code:

``` cs
public static class SynchronizeAsync
    {
        private static string serverConnectionString = AzureUrls;
        private static string clientConnectionString = DatabasePath();

        public static async Task Main() => await SynchronizeDataAsync();

        private static async Task SynchronizeDataAsync()
        {
            var serverProvider = new Dotmim.Sync.Web.Client.WebClientOrchestrator(serverConnectionString);

            var clientProvider = new SqliteSyncProvider(clientConnectionString);

            SyncFilter filter = new SyncFilter();
            filter.TableName = "Members";
            //filter ??

            // Creating an agent that will handle all the process
            var agent = new SyncAgent(clientProvider, serverProvider);

            // Launch the sync
            var result = await agent.SynchronizeAsync();
        }
    }`
```

Thank you for your help and advice in advance.

Kind regards,
Kamal

**Comment by Mimetis** [comment]
Did you see the [Xamarin Sample](https://github.com/Mimetis/Dotmim.Sync/tree/master/Samples/XamAppSync) ?

You can create an Azure Web Application to host the Xamarin Server Side, and access to your Azure Sql Instance

Try to make this sample working, and let me know if you have any problem

**Reply by sisaacks** [reply]
@Mimetis I have a very similar question, there is a very detailed "how to" on creating a web api for Azure.  There is also a detailed how to for deployment to Azure.  What I am confused about, in your instructions you say to create one controller.  The web api how to says to create a controller for each table.  I think I am asking what the guy did above, but I am not sure what the correct way to set up the api is with Dotmim
